pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java

```diff
@@ -1122,6 +1126,32 @@ public void writeBrokerDataOnZooKeeper(boolean force) {
         }
     }
 
+    /**
+     * sort bundles by load and select topK bundles for each broker.
+     * @return the number of bundles selected
+     */
+    private int selectTopKBundle() {
+        bundleArr.clear();
+        bundleArr.addAll(loadData.getBundleData().entrySet());
+
+        int maxNumberOfBundlesInBundleLoadReport = pulsar.getConfiguration()
```

Comment:
Currently, loadBalancerMaxNumberOfBundlesInBundleLoadReport is 10 by default. I wonder if this behavior change is accdptable. Shouldn't we get the Pulsar community's agreement to change this behavior? I still think it is safe to introduce a new flag to control this topk behavior for the modular load manager and by default it might need to be disabled.

Reply:
It is unworthy and weird that we introduce a new `boolean` type conf `enableBundleFilterForModularLoadManagerImpl` only for disabling this feature by default, as there are too many conf in Pulsar. 
And if you want to introduce a new `int` type conf `loadBalancerMaxBundleReportForModularLoadManagerImpl` to  avoid reusing  `loadBalancerMaxNumberOfBundlesInBundleLoadReport`, this is the original design in https://github.com/apache/pulsar/pull/20972, and we has discussed about this and close this PIP.
WDYT @dao-jun @lhotari @Technoboy- @codelipenghui @Demogorgon314 @heesung-sn

Reply:
I am worried about any possible regression caused by this behavior change. It would be great if we could double-confirm with the community. 

For example, this might cause a large cluster to take more time to balance load.

Reply:
It does, filtering too many bundles will degrade the performance of load balance. 
But i have a question, why do we set the default value of `loadBalancerMaxNumberOfBundlesInBundleLoadReport` to be 10? shouldn't we disable the topK filtering logic for `ExtensibleLoadManagerImpl` too? don't `ExtensibleLoadManagerImpl` have the simillar degradation problem of loadbalance when we filter most of the bundles?

Reply:
By introducing a new load balancer, we had more flexibility to drastically change the LB behavior, without worrying too much about the regression. (This is one of the reasons why we wanted to introduce a new LB not to cause regression in running clusters)

Reply:
This config does not take effect for modular load manager so I don't think it would cause regressions. But anyway, this change changes the semantic of a configuration so it's better to write a PIP for it.

See https://github.com/apache/pulsar/tree/master/pip#when-is-a-pip-required

> Any change to the configuration

Reply:
Hi, i create PIP-354(https://github.com/apache/pulsar/pull/22765) to describe about this improvement, and update the doc to notify those users who want to keep the same behavior of modular load manager that they can set the conf to be -1 to disable this feature. 
Hope that this can fix the problem, thanks!